lib_eio/file.ml

```diff
@@ -1,8 +1,47 @@
+(** Tranditional Unix permissions. *)
+module Unix_perm = struct
+  type t = int
+  (** This is the same as {!Unix.file_perm}, but avoids a dependency on [Unix]. *)
```

Comment:
i'm not so keen on this in the portable `eio` core, as we may just have very simple filesystem backends.  It would be nice to not have to dummy out `Stat.t` with fake values for uid/gid, for example.

Reply:
But this was in `Path`, `Path` is already part of the portable core, or did I misunderstand something ?

I'm unsure where to put the Stat, I was thinking on adding a new effect on `Eio_unix.Private`:
    `| Stat : <unix_fd> -> Stat.t Effect.t`

Does that make sense, technically all we need is a fd, or should I add to `Eio_unix.unix_fd` directly ?

Reply:
There's no problem with having the concept of "Unix permission" itself in the core - even non-Unix systems will often want to access Unix filesystems, tarballs, etc.

A more interesting question is whether we should add type parameters to `dir` to cover different types of file-system. e.g. our `dir` could become `Unix_perm.t dir`. However, that could be annoying, and any FS that can be mounted on Linux already has some suitable mapping.

Linux's `statx(2)` already allows you to do a stat that might or might not return some values (e.g. you can say you don't care about the UID and it is free to leave that field blank). So ideally we want a way to indicate which fields are provided even there.